thoth/common/openshift.py

```diff
@@ -1608,6 +1609,43 @@ def get_workflow(
 
         return wf
 
+    def get_workflow_status_report(
+        self,
+        workflow_id: Optional[str] = None,
+        label_selector: Optional[str] = None,
+        namespace: Optional[str] = None,
+    ) -> Tuple[Dict[str, Any], int]:
+        """Get workflow status report, derived from workflow status."""
+        if workflow_id is None:
+            return {"error": "No analysis id provided"}, 400
+        try:
+            wf_status = self.get_workflow_status(
+                name=workflow_id, label_selector=label_selector, namespace=namespace
+            )
+            status = {}
+            # the Job has not started (yet) but the Workflow has been submitted
+            # if the Workflow fails, the status will contain the finished
+            # time of the workflow
+            status["status"], code = (
+                {
+                    "finished_at": wf_status.get("finishedAt"),
+                    "reason": None,
+                    "started_at": wf_status.get("startedAt"),
+                    "state": wf_status.get("phase", "Pending"),
+                },
+                200,
+            )
+        except NotFoundException:
+            return (
+                {
+                    "error": f"Requested status for analysis {workflow_id!r} was not found"
+                },
+                404,
+            )
+        except Exception:
+            _LOGGER.error(status["error"])
```

Comment:
What should the function return in this case?  It will get to the next line and `status` and `code` have not been set.

Reply:
I think we should raise an exception here, WDYT?

Reply:
You don't need to do generic except block here. If any exception occurs, just let it be propagated.

Reply:
Will remove it :+1: